packages/shared/src/game_map.ts

```diff
@@ -2,9 +2,9 @@ import { Baduk } from "./baduk";
 import { AbstractGame } from "./abstract_game";
 import { Phantom } from "./phantom";
 
-// Is there any way we can get away from `any` here?
-// I think only the infra needs to touch these functions, but still!
-export const game_map: { [variant: string]: any } = {
+export const game_map: {
+  [variant: string]: new (config: any) => AbstractGame;
```

Comment:
This syntax is new to me. Does this mean that what comes after the colon must have a constructor that takes a config (of any type) and returns an AbstractGame?

Reply:
There's a lot of colons in here, but yeah I think that's right.

TBH the constructor syntax is new to me too.  I wanted to type this as `{ [variant: string]: AbstractGame }`, but then we run into the problem:

```
return new game_map[variant](config);
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    Cannot create an instance of an abstract class.(2511)
```